e2e/support/helpers/e2e-qa-databases-helpers.js

```diff
@@ -227,22 +228,39 @@ export function waitForSyncToFinish({
 
   cy.request("GET", `/api/database/${dbId}/metadata`).then(({ body }) => {
     if (!body.tables.length) {
-      waitForSyncToFinish({ iteration: ++iteration, dbId, tableName });
+      return waitForSyncToFinish({
+        iteration: ++iteration,
+        dbId,
+        tableName,
+        tableAlias,
+      });
     } else if (tableName) {
-      const hasTable = body.tables.some(
+      const table = body.tables.find(
         table =>
           table.name === tableName && table.initial_sync_status === "complete",
       );
-      if (!hasTable) {
-        waitForSyncToFinish({ iteration: ++iteration, dbId, tableName });
+
+      if (!table) {
+        return waitForSyncToFinish({
```

Comment:
Add a return statement to prevent the function call to continue after this point since we should only wrap `table` when the table value is not undefined (has value).